mirror.go

```diff
@@ -329,24 +331,48 @@ func (m *mirror) getRemoteTags() ([]RepositoryTag, error) {
 
 	var allTags []RepositoryTag
 	for {
-		req, err := http.NewRequest("GET", url, nil)
-		if err != nil {
-			return nil, err
-		}
+		var (
+			err     error
+			res     *http.Response
+			req     *http.Request
+			retries int = 5
+		)
+
+		for retries > 0 {
+			req, err = http.NewRequest("GET", url, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			if token != "" {
+				req.Header.Set("Authorization", fmt.Sprintf("JWT %s", token))
+			}
+
+			res, err = httpClient.Do(req)
+
+			if err != nil {
+				m.log.Warningf("Failed to get %s, retrying", url)
+				retries--
+			} else if res.StatusCode == 429 {
+				sleepTime := getSleepTime(res.Header.Get("X-RateLimit-Reset"), time.Now())
+				m.log.Infof("Rate limited on %s, sleeping for %s", url, sleepTime)
+				time.Sleep(sleepTime)
```

Comment:
Is it worth having `retries--` to avoid a potential for looping forever?

Reply:
retry with exponential growth of the sleep time?

Reply:
I decided to just decrement retries my reasoning being

- We're already sleeping for the duration that is provided from their API as to when the limit is reset
- This will exit with a failure after 5 attempts of that